pkg/worker/worker.go

```diff
@@ -76,6 +78,63 @@ type ContainerInstance struct {
 	Port         int
 	OutputWriter *common.OutputWriter
 	LogBuffer    *common.LogBuffer
+
+	memoryEvents MemoryEvents
+}
+
+func (i *ContainerInstance) isOOMKilled() bool {
+	podUID := os.Getenv("POD_UID")
+	if podUID == "" {
+		return false
+	}
+
+	memoryEventsPath := fmt.Sprintf("/sys/fs/cgroup/kubepods/besteffort/pod%s/%s/memory.events", podUID, i.Id)
```

Comment:
this ties us into the kubepods namespace pretty tightly. is there a way to make this more generic i.e. detect where this path is dynamically?

Reply:
I updated the cgroup path set on the runc spec to `/sys/fs/cgroups/<worker-id>/<container-id>`. Its unclear to me if this will work in k3s since processes are managed by systemd. I'll need to test a custom worker build in staging.